### DISCUSSION TITLE
page/erd-creator/

### DISCUSSION THREAD

**Discussion by giscus** [basvandesande/azurecodingarchitect#27] [General]
# page/erd-creator/

Tinkering with Azure and other technologies

https://azurecodingarchitect.com/page/erd-creator/

**Comment by RKeithHoward** [comment]
Hello Bas, Nice tool! Once you select your options, then generate a diagram, is there a way to reconfigure your options, and then regenerate? Maybe that already exists, but I had trouble finding it. Many thanks. -Keith

**Reply by basvandesande** [reply]
I have to think about the user workflow. Perhaps time for me to revamp the tool from scratch

**Comment by scankannan** [comment]
Hello Bas

I have a question regarding the .erd file that gets generated via the xrmtoolbox plugin, I would like to know what viewers can open this file, as if i needed to share a generated ERD file to other colleagues who do not have xrmtoolbox installed, what is the available option, i believe you must have used some diagraming tool library to render the diagram.

**Reply by basvandesande** [reply]
Sorry, I missed your comment. The purpose of the ERD creator is to create images. In the toolbar you see a small photo camera icon. If you click that button an image will be created that you can use in your documents. I also added an experimental feature to generate a draw.io chart (but that one is still buggy)

**Comment by Isqandar01** [comment]
Hi, thank for the tool. 
However, recently there's a bug preventing me to close the Welcome Page.
The Close button at the bottom is missing/hidden from view. So I cant close it.

![Welcome Page](https://i.imgur.com/dOMgSAr.png)

**Reply by SweetLive** [reply]
Hello, I have same issue. Do we have an update to fix it plz ?

**Reply by Isqandar01** [reply]
solved it.. in my case, this issues was due to WIndows Display settings.
Desktop > Right Click > Display Seetings > Display > Scale and Layout

Initially my setting was 125% scale, so I changed it to 100% and the Close Button is no longger hidden

**Reply by lm00750** [reply]
I have had this issue and got round it by using the tab and enter keys.
Tab > enter

**Reply by basvandesande** [reply]
Sorry I missed your comments. I'll see what I can do about that. I have to see how I can tackle the scaling issue with classic winforms and 4K screens

**Reply by McGern** [reply]
I've got the same. Thought at first that the program had crashed, and I pressed alt+f4 in an attempt to close the program, but it ended up closing just the overlay which was handy

**Comment by MatthiasSomay** [comment]
Hi Bas, 

Is there a way to delete relationship lines without also deleting the linked tables? Right now every entity has a lookup to the Teams tabel. In the case of a lot of tables this can result in too many lines. For illustration purposes this can be handy. Thanks for the great tool!

**Reply by basvandesande** [reply]
Hi Matthias, 

if I'm correct you can select the relationship, then use the right mouse button to delete it.

Bas

**Comment by iddycol** [comment]
Hi Bas,

When working in the context of a solution I'd like to select one of the entities displayed and focus in on only that entity and everything I have related to it. 

Is that possible? 

I figured I could use the load entities tab and select just the entities I want manually but that's super tedious and could be prone to errors if the relationships are not what I am expecting.

Thoughts?

Adrian

**Comment by VijayAT123** [comment]
Haven't been able to find a tool that parallels the ease and functionality of this, great work!

Feature request: hovering over a field will display the friendly name, logical name, type, and description from Power Apps.

**Comment by cbuctok** [comment]
Hello Bas, the Entity Relationship Diagram Creator plug-in a nifty tool for visualizing data models in Dataverse!
Cannot imagine designing/supporting complex solutions without it.
Quick question: which tool should I use to view ERDs? I tried DBeaver, but it keeps saying “Diagram version not found.” Any tips?
Thank you again!
Greg

**Comment by jpvj** [comment]
Hi!

I just found your tool and started playing around with it.
Thank you very much for the time spend - It's a nice tool that will help me a lot!

I have a feature request (or two): 
After rearranging/collapsing my entities I need to add another entitry or remove it. When pressing "Generate" I have to start from scratch with my own layout based on the grid-layout.
Would you consider to "keep the layout changes" and just trying to add/remove any entities?

If possible, make a option to have the relationship lines not drawn on top of each other. I know it will require some more logic, but it would be a great improvement.

I found a few bugs - how do you want me to report them?
1) Selecting "Custom fields from all publishers" causes an exception in my setup.
2) Moving an entity suddently causes the relation lines to be extended very far below the entities.

Best regards and thank you!

/JP

**Comment by StephanieLoomans** [comment]
Hey :) I read very good thing about your tool but it appears it is not available on XRM toolbox any more? is this correct? Could it perhaps be available through another channel?

**Reply by McGern** [reply]
I've just downloaded xrmtoolbox and managed to add this plugin from Configuration=>Tool library

Trying to upload an image, but can't - from the Tool Library I typed "Entity Relation" in the search which finds the plugin
